api_v2/src/leaves/leaves.service.ts

```
@@ -161,6 +173,361 @@ export class LeavesService {
+    const notifications = await this.prisma.notification.findMany({
+      where: {
+        relatedEntityId: leaveId,
+        recipientId: projectLeaderId,
+        OR: [{ type: 'leave' }, { type: 'undertime' }],
+      },
+    });
+
+    for (const notification of notifications) {
+      try {
+        let parsedData: any = {};
+        try {
+          parsedData =
+            typeof notification.data === 'string'
+              ? JSON.parse(notification.data)
+              : notification.data;
+        } catch (e) {
+          continue;
+        }
+
+        // Update the Status field
+        parsedData.Status = isApproved
+          ? RequestStatusEnum.APPROVED
+          : RequestStatusEnum.DISAPPROVED;
+
+        // Determine notification type
+        const isUndertime =
+          notification.type === 'undertime' ||
+          notification.type === 'undertime_resolved';
+        const newType = isApproved
+          ? isUndertime
+            ? 'undertime_resolved'
+            : 'leave_resolved'
+          : notification.type;
+
+        await this.prisma.notification.update({
+          where: { id: notification.id },
+          data: {
+            data: JSON.stringify(parsedData),
+            type: newType,
+          },
+        });
+      } catch (e) {
+        // Continue on error
+      }
+    }
+  }
+
+  private async updateNotificationData(
+    leaveId: number,
+    approvingUserId: number,
+    isApproved: boolean,
+  ): Promise<void> {
+    const notifications = await this.prisma.notification.findMany({
+      where: {
+        relatedEntityId: leaveId,
+        recipientId: approvingUserId,
+      },
+    });
+
+    for (const notification of notifications) {
+      try {
+        let parsedData: any = {};
+        try {
+          parsedData =
+            typeof notification.data === 'string'
+              ? JSON.parse(notification.data)
+              : notification.data;
+        } catch (e) {
+          continue;
```

Add error message

The code swallows errors silently, making debugging impossible if something fails (like malformed JSON or a database issue).

- Silent error swallowing in notification update methods: catch (e) { continue }
- Makes debugging difficult and hides potential issues
- Recommendation: Log errors or use proper error handling